### ISSUE TITLE
Permalinks for hacks

### ISSUE THREAD

**Issue by ches** [bkkhack/hackmap#18]
Related to #7, permalinks for each hack night event:

> As a hacker I want a direct link to view my hack, so that I can share my plan with others and invite them to join me.

Ideally I suppose we could use `pushState` so that when someone clicks to view their own hack they can simply copy the URL from the address bar. Could be something like:

https://bkkhack.github.io/hackmap/events/12/hacks/325548694

where 325548694 is the GitHub comment ID—I'm assuming that's what it is from URIs like this:

`https://github.com/bkkhack/hackmap/issues/12#issuecomment-325548694`
https://github.com/bkkhack/hackmap/issues/12#issuecomment-325548694

**Comment by ches** [comment]
https://router.vuejs.org/en/

**Comment by djay** [comment]
@ches given there should only be one comment per person, using the github id might make for a nicer url.

**Comment by ches** [comment]
Thinking about it, not sure that relying on or enforcing one comment/hack per person would be desirable: there are times when I think of posting two ideas I could work on, and seeing which gets interest from others.

**Comment by waf** [comment]
Yeah, it's interesting because we've coupled physical location with what you're hacking on. It makes sense to have more than one hack, but not to have more than one physical location. 

I think the benefit of the coupling is that there's a very low learning curve for hackmap. In the past I've put multiple ideas in the hack description, which seems to work out OK.

**Comment by djay** [comment]
> On 24 Jan 2018, at 8:44 pm, Will Fuqua <notifications@github.com> wrote:
> 
> Yeah, it's interesting because we've coupled physical location with what you're hacking on. It makes sense to have more than one hack, but not to have more than one physical location.
> 
> I think the benefit of the coupling is there's a very low learning curve for hackmap. In the past I’ve put multiple ideas in the hack description, which seems to work out OK.

I like the concept a person place who has a bunch of interests who can then move around. It’s simple to understand.
The way I thought about it is that if multiple people work on the same thing then they all just change their status to the same project idea. We couuld extend the project idea to a link to another issue or repo…

But there are a couple of other ways to do things. We could have both projects and people on the map and both could move around.

Or if there is just projects/hacks that have icons on them to represent who has joined that project.

It’s more a question of which way encourages people to participate, communicate with each other and encourage more people to join earlier. And which way requires the least explanation.
